core/src/Kokkos_View.hpp

```diff
@@ -132,6 +132,66 @@ void runtime_check_rank_host(const size_t dyn_rank, const bool is_void_spec,
 }
 #endif
 
+template <typename Traits, typename Layout>
+KOKKOS_INLINE_FUNCTION void runtime_check_rank(
+    Layout& layout, const std::string& label) {
+  size_t i0 = layout.dimension[0];
+  size_t i1 = layout.dimension[1];
+  size_t i2 = layout.dimension[2];
+  size_t i3 = layout.dimension[3];
+  size_t i4 = layout.dimension[4];
+  size_t i5 = layout.dimension[5];
+  size_t i6 = layout.dimension[6];
+  size_t i7 = layout.dimension[7];
+
+#ifdef KOKKOS_ENABLE_OPENMPTARGET
+  KOKKOS_IMPL_IF_ON_HOST
```

Comment:
after the #4660 is in you should use that.

Reply:
Yes - @dalg24, please rebase #4660 once that PR is ready so we catch this ifdef and update.